api/api.yaml

```diff
@@ -1352,6 +1352,270 @@ paths:
+                          valuesRestricted: true
+              createNumericFields:
+                summary: Create numeric fields with constraints
+                value:
+                  fields:
+                    Opportunity:
+                      - fieldName: "Deal_Score"
+                        displayName: "Deal Score"
+                        description: "Likelihood score from 0-100"
+                        valueType: "number"
+                        required: true
+                        numericOptions:
+                          precision: 5
+                          scale: 2
+                          min: 0
+                          max: 100
+                          defaultValue: 50
+              createAssociationField:
+                summary: Create field with association to another object
+                value:
+                  fields:
+                    Task:
+                      - fieldName: "Related_Contact"
+                        displayName: "Related Contact"
+                        valueType: "reference"
+                        association:
+                          associationType: "lookup"
+                          targetObject: "Contact"
+                          targetField: "Id"
+                          cardinality: "many-to-one"
+                          required: false
+                          labels:
+                            singular: "Contact"
+                            plural: "Contacts"
+      responses:
+        '200':
+          description: Metadata creation completed successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/UpsertMetadataResponse'
+        '400':
+          description: Invalid request payload or validation errors
+          content:
+            application/problem+json:
+              schema:
+                $ref: "../problem/problem.yaml#/components/schemas/ApiProblem"
+        '401':
+          description: Authentication required
+          content:
+            application/problem+json:
+              schema:
+                $ref: "../problem/problem.yaml#/components/schemas/ApiProblem"
+        '403':
+          description: Insufficient permissions for metadata operations
+          content:
+            application/problem+json:
+              schema:
+                $ref: "../problem/problem.yaml#/components/schemas/ApiProblem"
+        '404':
+          description: Project, integration, or installation not found
+          content:
+            application/problem+json:
+              schema:
+                $ref: "../problem/problem.yaml#/components/schemas/ApiProblem"
+        '500':
+          description: Internal server error
+          content:
+            application/problem+json:
+              schema:
+                $ref: "../problem/problem.yaml#/components/schemas/ApiProblem"
+  /projects/{projectIdOrName}/objectMetadata:
```

Comment:
this path should be `/projects/{projectIdOrName}/providers/{provider}/objectMetadata` (provider in path, not query param) to align with https://docs.withampersand.com/reference/objects-&-fields/get-object-metadata-for-connection